controllers/spacerequest/spacerequest_controller_test.go

```diff
@@ -175,6 +243,71 @@ func TestCreateSpaceRequest(t *testing.T) {
 				HasSpecTargetClusterRoles([]string(nil)). // has empty target cluster roles
 				HasSpecTargetCluster("member-2")          // subSpace has same target cluster as parentSpace
 		})
+
+		t.Run("space request has already an existing secret", func(t *testing.T) {
+			// given
+			kubeconfigSecret1 := test.CreateSecret("jane-xyz1", sr.Namespace, map[string][]byte{
+				"kubeconfig": []byte(fakeKubeConfigSecret()),
+			})
+			kubeconfigSecret1.StringData = map[string]string{
+				"kubeconfig": fakeKubeConfigSecret(),
+			}
```

Comment:
this looks like duplication of the same :thinking:

Reply:
yes, in that case it sets the kubeconfig in both fields because the `CreateSecret` function sets the `secret.Data` field, while the spacerequest_controller and now also `HasNamespaceAccess` test utility checks the `secret.StringData` field. 
So yeah, maybe it can be cleaned up.